driver-core/src/main/java/com/datastax/driver/core/ChainedResultSetFuture.java

```diff
@@ -0,0 +1,71 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.core;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import com.google.common.util.concurrent.AbstractFuture;
+import com.google.common.util.concurrent.FutureCallback;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.Uninterruptibles;
+
+/**
+ * A {@code ResultSetFuture} that will complete when its source future completes.
+ */
+class ChainedResultSetFuture extends AbstractFuture<ResultSet> implements ResultSetFuture {
```

Comment:
Unfortunately not.
The catch is that, when we build the `ChainedResultSetFuture`, the "source" future does not exist yet. So it's not exactly a delegate.
See `SessionManager#executeAsync`.